server/server.go

```diff
@@ -876,6 +878,48 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	return &httpS
 }
 
+var extensionsPattern = regexp.MustCompile(`^extension(.*)\.js$`)
+
+func (a *ArgoCDServer) serveExtensions(extensionsSharedPath string, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "application/javascript")
+
+	err := filepath.Walk(extensionsSharedPath, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to iterate files in '%s': %w", extensionsSharedPath, err)
+		}
+		if !info.IsDir() && extensionsPattern.MatchString(info.Name()) {
```

Comment:
This is probably paranoid, but how would you feel about skipping symlinks? The endpoint is a little scary because it has no auth and will dutifully just write files to a response. An attack would require _getting_ a symlink into the extensions directory, but the blast radius could be pretty huge.

Reply:
A little bit of paranoia won't harm :) 

I agree it is better to be careful here. Added symlink check as well.